src/quo2/components/community/token_gating.cljs

```diff
@@ -1,295 +1,45 @@
-                                    :align-items    :center}
-   :token-row-container            {:flex-direction    :row
-                                    :align-items       :center
-                                    :margin-horizontal (- token-tag-horizontal-spacing)}
-   :token-row-container-no-shift   {:margin-left 0}
-   :token-row-or-text              {:margin-right 3}
-   :token-tag-spacing              {:margin-horizontal token-tag-horizontal-spacing
-                                    :margin-vertical   token-tag-vertical-spacing}
-   :divider                        {:height            1
-                                    :margin-horizontal -20
-                                    :margin-top        (- 12 token-tag-vertical-spacing)
-                                    :margin-bottom     8}
-   :membership-request-denied      {:margin-top 13}
-   :enter-button                   {:margin-top 16}
-   :info-text                      {:margin-top         12
-                                    :text-align         :center
-                                    :padding-horizontal 20}})
-
-(defn multiple-token-requirements?
-  [token-requirement-lists]
-  (when (vector? (first token-requirement-lists))
-    (> (count token-requirement-lists) 1)))
-
-(defn is-token-requirement-met?
-  [token-requirement-list]
-  (and (seq token-requirement-list)
-       (every?
-        (fn [token]
-          (get token :is-sufficient?))
-        token-requirement-list)))
-
-(defn are-multiple-token-requirements-met?
-  [token-requirement-lists]
-  (if (multiple-token-requirements? token-requirement-lists)
-    (some is-token-requirement-met? token-requirement-lists)
-    (is-token-requirement-met? token-requirement-lists)))
+            [clojure.string :as string]
+            [quo2.components.community.style :as style]))
 
 (defn token-requirement-list-row
-  [tokens community-color]
+  [tokens padding?]
   [rn/view
-   {:style (merge
-            (get styles :token-row)
-            (get styles :token-requirement-list-spacing))}
-   (map-indexed (fn [token-index token]
-                  (let [{:keys [token-img-src token amount is-sufficient? is-purchasable?]} token]
-                    ^{:key token-index}
-                    [rn/view {:style (get styles :token-tag-spacing)}
-                     [token-tag/token-tag
-                      {:token          token
-                       :value          amount
-                       :size           24
-                       :border-color   community-color
-                       :is-required    is-sufficient?
-                       :is-purchasable is-purchasable?
-                       :token-img-src  token-img-src}]]))
-                tokens)])
+   {:style (style/token-row padding?)}
+   (doall
+    (map-indexed (fn [token-index token]
+                   (let [{:keys [img-src symbol amount sufficient? purchasable? loading?]} token]
+                     ^{:key (str "token" token-index)}
```

Comment:
no need to use `(str "token"` as keys need to be unique per array, so just `index` would be the same (it's also the default to use the index)

What about symbol, would that be unique?

Reply:
it's not my code, but im fine with any token, i don't think it matters much? str "token" is fine for me, why change ?

Reply:
it's considerably slower, it might not matter here, but then people copy over the pattern and might use it in areas where it matters, so I would try not to use it anywhere

Reply:
makes sense, thanks